### PR TITLE
Remove `StepIcon` in `StepStatus` and use `@yamada-ui/lucide` icon

### LINKED ISSUE #3537: Replace `StepIcon` with `LucideIcon`
### Description

Replace `StepIcon` with `LucideIcon`

### Problem Statement/Justification

Replace the icons from `@yamada-ui/icon` with those from `@yamada-ui/lucide`.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by taroj1205**
Closes #3537 

## Description

Remove `StepIcon` in `StepStatus` and use `@yamada-ui/lucide` icon.

### CODE REVIEW

**Review by illionillion — APPROVED** [review]
LGTM

**Review by mei28 — APPROVED** [review]

**Comment by mei28** [comment]
Sorry, I accidentally approved it again. I think it's fine.

File: packages/components/stepper/src/step-status.tsx
Comment (hirotomoyamada) [inline packages/components/stepper/src/step-status.tsx:1]: @taroj1205 @mei28 @illionillion 

Hi, You can't apply the `styles.icon` style.

Comment (taroj1205) [reply packages/components/stepper/src/step-status.tsx:1]: Hi @hirotomoyamada, what do you reckon I should be changing?
Should I remove icon from step.ts in styles or wrapping the icon in a div with `__css={iconStyles}` as a prop?

Comment (hirotomoyamada) [reply packages/components/stepper/src/step-status.tsx:1]: It looks like it would need to be wrapped in a `div` tag to inherit the style from the theme.

Comment (taroj1205) [reply packages/components/stepper/src/step-status.tsx:1]: Hi @hirotomoyamada, I've wrapped it in a div, please check!

Comment (hirotomoyamada) [reply packages/components/stepper/src/step-status.tsx:1]: No, why not just bring back `StepIcon` and replace the `CheckIcon` used internally?

**Pull request closed without merging**